### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by magdalenakierat** [neuefische/questions#60] [Web] [answered]
Hallo zusammen! 
Ich möchte die Card Component testen und bekomme die Fehlermeldung `useHref() may be used only in the context of a <Router> component`. Ich dachte, ich könnte es mit `MemoryRouter` lösen (ähnlich wie bei Storybook) und bekomme nun diese Fehlermeldung:  `TypeError: Cannot read properties of undefined (reading 'pathname')`.
Hier der Link zum entsprechenden Branch: https://github.com/magdalenakierat/capstone-project/compare/testing?expand=1
Danke für jeden Hinweis! 🙏

**Comment by mbosselmann** [comment, marked as answer]
Hallo Magda,
in deine `Card.js` reichst du einen **pathname** rein. Deswegen brauchst du in deinem Test auch einen **pathname**.
Wenn du in deiner `Card.spec.js` `<Card pathname="pathname wert von dir hier einfügen" />` schreibst, sollte es gehen.
Ich hoffe, dass hilft dir weiter! :-)

**Comment by magdalenakierat** [comment]
Danke, liebe @mbosselmann, so klappt es! 🎉
